aptos-move/e2e-move-tests/src/tests/v5_module_v1_metadata.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, assert_vm_status, MoveHarness};
+use aptos_cached_packages::aptos_stdlib;
+use aptos_framework::{BuildOptions, BuiltPackage};
+use aptos_package_builder::PackageBuilder;
+use aptos_types::{account_address::AccountAddress, on_chain_config::FeatureFlag};
+use move_binary_format::CompiledModule;
+use move_core_types::{metadata::Metadata, vm_status::StatusCode};
+use serde::Serialize;
+use std::collections::BTreeMap;
+
+#[test]
+fn test_v5_module_v1_metadata() {
```

Comment:
I guess this not intended to land?

Tests like this should not be merged, because they are unmaintainable. Previously when we only had bytes, we translated them back to code building a compiled program. There are tools for this.

Reply:
actually I think this is perfectly acceptable

I mean we could regenerate the source for how we got here, but I think have corpus of what should be static data is fine.

If this becomes flaky in the future, then we have changed our bytecode processors into something incompatible.

The issue we had in the past is with goldens on a set of resources that are not fixed.

A lot of applications incorporate a corpus of well known data for validation of correctness.

Reply:
probably should at least store the corpus as separate files and have the source code/instructions on how to generate the corpus?

Reply:
Given the amount of time to further improve this, I think it is easier to just delete it and we can revisit this later. The fundamental issue is that we need better feature gating around bytecode versions and metadata. We probably should also fail any publish transaction that has any other type of module metadata. Might be too late, but we should add that as a feature gate to prevent code complexity increasing too much.